vaxrank/cli.py

```diff
@@ -190,38 +197,108 @@ def load_template_data(args):
         min_epitope_score=args.min_epitope_score)
 
     ranked_list_for_report = ranked_list[:args.max_mutations_in_report]
-    df = dataframe_from_ranked_list(ranked_list_for_report)
-    logger.debug(df)
+    data = {
+        'variants': ranked_list_for_report,
+        'saved_args': args,
+    }
+    logger.info('About to save args: %s', data['saved_args'])
+
+    # save pickle data if necessary. as of time of writing, vaxrank takes ~25 min to run,
+    # most of which is core logic. the formatting is super fast, and it can
+    # be useful to save the data to be able to iterate just on the formatting
+    if args.output_pickle_file:
+        with open(args.output_pickle_file, 'wb') as f:
+            pickle.dump(data, f)
```

Comment:
Now that FMIndex on py3 is fixed (amazing turnaround time on that shellinford issue!), pickling in py3 fails at serialization with:

```
Traceback (most recent call last):
  File "/Users/julia/Envs/vaxrank3/bin/vaxrank", line 11, in <module>
    load_entry_point('vaxrank', 'console_scripts', 'vaxrank')()
  File "/Users/julia/code/vaxrank/vaxrank/cli.py", line 333, in main
    data = ranked_variant_list_with_saved_args(args)
  File "/Users/julia/code/vaxrank/vaxrank/cli.py", line 216, in ranked_variant_list_with_saved_args
    cPickle.dump(data, f)
AttributeError: Can't pickle local object 'combine_scoring_functions.<locals>.ManufacturabilityScores'
```

Unsurprisingly, this fails independent of protocol. So the issue is that pickling doesn't play nice with `ManufacturabilityScores`, and `serializable` doesn't play nice with `argparse` objects. At this point, there are a couple of options:
- make ManufacturabilityScores an actual namedtuple or class, and not construct it on the fly (stick with pickling)
- don't save args from a previous run, when reloading
- look into rewrite `mhc_alleles_from_args`, `variant_collection_from_args` to be constructed from dicts instead of `argparse.Namespace` objects - this probably wouldn't be so bad - and switch from pickling to `serializable` (kinda overloading `serializable` some more)

@alex thoughts?